src/RA_AchievementOverlay.cpp

```diff
@@ -74,22 +68,22 @@ const unsigned int OVERLAY_HEIGHT = 1024;
 
 void AchievementOverlay::SelectNextTopLevelPage(BOOL bPressedRight)
 {
-    switch (m_Pages[m_nPageStackPointer])
+    switch (m_Pages.at(m_nPageStackPointer))
     {
-        case OP_ACHIEVEMENTS:
-            m_Pages[m_nPageStackPointer] = (bPressedRight ? OP_FRIENDS : OP_LEADERBOARDS);
+        case Page::Achievements:
+            m_Pages.at(m_nPageStackPointer) = (bPressedRight ? Page::Friends : Page::Leaderboards);
```

Comment:
`at `won't throw an exception with `std::array` because it is not a polymorphic container.